.github/workflows/design-system-ci.yml

```diff
@@ -0,0 +1,58 @@
+name: Design System CI and QA
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - staging-next
+        paths:
+            - 'frontend/**'
```

Comment:
Do we need this on `push` to those branches? Who is checking for failures? What actions should they take?

Reply:
Yes, in the next PR, the `push` trigger will initiate a CHANGELOG, version bump, and github release for `@code-dot-org/dsco`. This would only be triggered if files changed under the glob `frontend/**/*`. 

For failures, I plan to add (in a followup PR), notifications to be sent to the `#infra-cms-[env]` channels.

Reply:
Hmm, I have some concerns about intermingling pre-merge actions and post-merge actions, but I'll wait until I see what that looks like.

Reply:
I think it makes sense to separate the workflow, I'll update the PR to do that.